app/Filament/Resources/UserResource/Pages/Roles.php

```diff
@@ -37,16 +43,62 @@ public function table(Table $table): Table
                 Tables\Actions\AttachAction::make()
                     ->label(__('Add'))
                     ->color('primary')
-                    ->authorize(fn () => auth()->user()->can('updateRoles', $this->getRecord()))
+                    ->authorize(fn () => $user->can('updateRoles', $this->getRecord()))
                     ->recordTitle(fn (Model $record) => __('permission.role.' . $record->name))
                     ->preloadRecordSelect()
-                    ->recordSelectOptionsQuery(fn (Builder $query) => $query->whereIn('name', auth()->user()->assignableRoles)),
+                    ->recordSelectOptionsQuery(fn (Builder $query) => $query->whereIn('name', $user->assignableRoles))
+                    ->after(function ($data) {
+                        /** @var User $targetUser */
+                        $targetUser = $this->getRecord();
+
+                        $attachedRole = Role::findById((int) $data['recordId']);
+
+                        $newPermissions = Permissions::Registered;
```

Comment:
I believe having this default will strip away Developer permissions if a user is being assigned a secondary role (like Writer)

Reply:
Should be addressed both here and in the detachment action in latest.